Google.PowerShell.Tests/Common/TestActiveUserCredential.cs

```diff
@@ -0,0 +1,138 @@
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            string accessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+
+            // The access token returned by GetAccessTokenForRequestAsync should come from token.
+            Assert.IsTrue(
+                Equals(activeUserToken.AccessToken, accessToken),
+                "GetAccessTokenForRefreshAsync returns the wrong access token.");
+
+            // The next call to GetAccessTokenForRequestAsync should returns the same token.
+            accessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+            Assert.IsTrue(
+                Equals(activeUserToken.AccessToken, accessToken),
+                "GetAccessTokenForRefreshAsync returns the wrong access token.");
+        }
+
+        /// <summary>
+        /// This test checks that GetAccessTokenForRequestAsync returns a new token
+        /// when the old one expired due to time.
+        /// </summary>
+        [Test]
+        public void TestGetAccessTokenExpiredByTime()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            // Make the token expired by time.
+            activeUserToken.Issued = DateTime.Now.AddSeconds(-activeUserToken.ExpiresInSeconds.Value);
+            Assert.IsTrue(activeUserToken.IsExpired(SystemClock.Default), "ActiveUserToken should be expired");
+
+            string newAccessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+            Assert.IsFalse(
+                Equals(activeUserToken.AccessToken, newAccessToken),
+                "GetAccessTokenForRefreshAsync should returns a new access token when the old one expired due to time."
+                );
+
+            // The next call to GetAccessTokenForRequestAsync should returns the same token.
+            string anotherNewAccessToken = activeUserCred.GetAccessTokenForRequestAsync(null, CancelToken).Result;
+            Assert.IsTrue(
+                Equals(anotherNewAccessToken, newAccessToken),
+                "GetAccessTokenForRefreshAsync should returns a new access token when the old one expired due to time."
+                );
+        }
+
+        /// <summary>
+        /// This test checks that GetAccessTokenForRequestAsync returns a new token
+        /// when the active user is changed.
+        /// </summary>
+        [Test]
+        public void TestGetAcessTokenExpiredByChangingUser()
+        {
+            ActiveUserCredential activeUserCred = new ActiveUserCredential();
+            ActiveUserToken activeUserToken = TokenProperty.GetValue(null, null) as ActiveUserToken;
+
+            // Make the token expired by changing the active user
```

Comment:
This comment and the error message below (line 121) aren't 100% correct. It isn't that the token has expired (ran out of time). It is simply no longer valid. By changing the name, the hash/whatever it uses to validate that token is no longer correct. Perhaps we sould just say "Invalidate the token by changing the active user."?